src/modules/categories/categories.service.ts

```
@@ -0,0 +1,201 @@
+      const category = await this.entityManager.findOne(Category, {
+        where: { id },
+      });
+      if (!category) {
+        return APIResponse.error(
+          res,
+          `Category with ID ${id} not found`,
+          "ERROR_DELETE_CATEGORY",
+          "Error Deleting Category",
+          HttpStatus.NOT_FOUND
+        );
+        throw new NotFoundException(`Category with ID ${id} not found`);
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)